### PR TITLE
list builders- adding "overrideOptionStyle" prop

### PR DESCRIPTION
**Pull request by TalZa1810**
### 🔦 Summary
This PR includes:
1. Adding `overrideOptionStyle ` prop to list Item builders to prevent adding extra styles from the dropdownLayout option itself (except `badgeSelectItemBuilder`- see comment below).
2. Adding / fixing the type files accordingly. 
3. Adding visual tests.
4. Adding deprecation logs for `title` and `divider` old API usage
5. Adding deprecation notation to `overrideStyle` and `title` API description. 

### ✅ Checklist

- [ ] 👨‍💻 API change is approved by the UI Infra Developers 
- [ ] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [ ] Story
  - [x] API description
  - [ ] Cheatsheet
  - [ ] Other (explain)
- 🔬 Change is tested
  - [ ] Component
  - [x] Visual test

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6151

**Review by shlomitc — COMMENTED** [review]
Now that you use `overrideOptionStyle ` in the builders, you can remove the `overrideStyle` usage from them

**Comment by TalZa1810** [comment]
Currently, the prop `overrideOptionStyle` was not added to `BadgeSelectItem` since it has no hovered, selected and disabled styles. These styles come from the DDL. I've created a Jira dedicated to this issue:
https://jira.wixpress.com/browse/DSM-955

**Comment by TalZa1810** [comment]
This fix removes the `selected` styles from the [ListItemEditable](https://github.com/wix/wix-style-react/pull/6122). 
FYI @bamai

File: src/DropdownLayout/DropdownLayout.js [hunk 1]
```diff
@@ -441,21 +446,22 @@ class DropdownLayout extends React.PureComponent {
       <div
         {...this._getItemDataAttr({ ...optionState, overrideStyle })}
         className={st(classes.option, {
+          overrideOptionStyle,
```
Comment (shlomitc) [inline src/DropdownLayout/DropdownLayout.js:449]: why not simply remove the classes?
I think it's less readable to add `:not()` in the css everywhere.
simply remove the class from the dom

**Comment by TalZa1810** [comment]
fails on `BarChart` visual test (not related).

**Pull request merged**